src/actions.ml

```diff
@@ -28,28 +41,37 @@ let send_status_check ~bot_info job_info pr_num (gh_owner, gh_repo)
         GitHub_queries.get_pull_request_refs ~bot_info ~owner:gh_owner
           ~repo:gh_repo ~number
         >>= function
-        | Ok {issue= id; head; last_commit_message= Some commit_message}
+        | Ok {issue= id}
         (* Commits reported back by get_pull_request_refs are surrounded in double quotes *)
-          when String.equal head.sha (f "\"%s\"" job_info.commit) ->
-            GitHub_mutations.post_comment ~bot_info ~id
-              ~message:
-                (f
-                   "For your complete information, the job \
-                    [%s](https://gitlab.com/%s/-/jobs/%d) in allow failure \
-                    mode has failed for commit %s: %s%s"
-                   job_info.build_name repo_full_name job_info.build_id
-                   job_info.commit
-                   (first_line_of_string commit_message)
-                   ( if
-                     String.equal job_info.build_name
-                       "library:ci-fiat_crypto_legacy"
-                   then "\nping @JasonGross"
-                   else "" ))
-        | Ok {head} ->
-            Lwt_io.printf
-              "We are on a PR branch but the commit (%s) is not the current \
-               head of the PR (%s). Doing nothing.\n"
-              job_info.commit head.sha
+          ->
+            let message =
+              f "The job [%s](%s) has failed in allow failure mode%s"
+                job_info.build_name job_url
+                ( if
+                  String.equal job_info.build_name
+                    "library:ci-fiat_crypto_legacy"
+                then "\nping @JasonGross"
+                else "" )
+            in
+            exec_github_app_or_not ~bot_info
+              ~github_app_func:
+                ( GitHub_queries.get_repository_id ~bot_info ~owner:gh_owner
+                    ~repo:gh_repo
+                >>= function
+                | Ok repo_id ->
+                    GitHub_mutations.create_check_run ~bot_info ~name:context
+                      ~repo_id ~head_sha:job_info.commit ~conclusion:NEUTRAL
+                      ~status:COMPLETED
+                      ~title:(failure_reason ^ " on GitLab CI")
+                      ~text:"" ~details_url:job_url ~summary:message
+                | Error e ->
+                    Lwt_io.printf "No repo id: %s\n" e )
+              ~github_access_token_func:
+                ( if
+                  String.equal job_info.build_name
+                    "library:ci-fiat_crypto_legacy"
+                then GitHub_mutations.post_comment ~bot_info ~id ~message
+                else Lwt.return () )
```

Comment:
This is executed only in the `github_access_token` case when it is in fact something very specific to the Coq repository itself, which uses the GitHub App.